### ISSUE TITLE
XSLT 3

### ISSUE THREAD

**Issue by Julegirl** [ebeshero/DHClass-Hub#567]
I am so stuck. I was able to get my headings just fine, but I can't get the contents to go where they are supposed to. I don't know what to do or say for the output. I know that <tr> represents tabele row and <td> represents a new column for the table and I know how to put stuff there, but I do I put in the right contents?

**Comment by ebeshero** [comment]
Hi @Julegirl ! I just added some of those tick marks to your code so we can read your post more clearly.  Okay, so just start by seeing if you can set up your template rules in the right place.  I think @BMT45 was stuck where you are yesterday, so take a look at the issue he posted and our responses: https://github.com/ebeshero/DHClass-Hub/issues/566

**Comment by ebeshero** [comment]
@Julegirl Let me see if I can help a little more here, too. You mentioned you have your table headings just fine, so that tells me you've got your template rule that matches on the document node, and you're outputting the `<table>` element and its top header row, right? 

Now, you want to be able to **select** what part of the tree you want to process for the rest of that table. This is where you want to use `<xsl:apply-templates select="descendant::fs[the-ones-with-the-yes-answers]"/>`, like this:

```
<xsl:template match="/">
<html>
      <head><title>STUFF</title></head>
<body>
<table>
     <tr> 
           <th>.....TABLE HEADER STUFF....</th>
     <tr>
   <xsl:apply-templates select="select="descendant::fs[the-ones-with-the-yes-answers]"/>
<table>
</body>
</html>
</xsl:template>
```

You need a NEW template rule next that will match on those fs elements that have "Yes" answers in them. Every time that template rule matches something in the tree, you want to output a new table row containing its data. Does that help? (Also see the issue @BMT45 posted for more sample code).

**Comment by Julegirl** [comment]
Yeah, I got that far, but I don't know where to go from here. It's just that how you worded everything i confusing me greatly.

**Comment by ebeshero** [comment]
@Julegirl Quote where you're confused, and I'll see if I can improve it...?

**Comment by Julegirl** [comment]
It's #3 and 4
 In this case, then, your <xsl:apply-template> elements inside the template rule for the document node will tell the system that you want to process the descendant <fs> elements, based on their contents. In order to specify that you only want specific Yes or No Questions then you need to specify with <xsl:apply-template select="___"> to process fs elements only when one of the <f name="response"> has an @select="Yes". That work wll get done by new <xsl:template> rules that you will need to write to match the <f name="response"> with @select="Yes".
In the <xsl:template> rules for <string> elements of the <f name="question"> you’ll need to output something for each one, that is, each question that has a “Yes” answer (held inside the same <fs> element as the <f name="response"> with @select="Yes").

I just don't understand what your asking, or how I'm supposed to do it.

**Comment by ebeshero** [comment]
@Julegirl Okay, I pretty much was writing about exactly that in this issue here on DHClass-Hub. Take a look at the code block I posted--the point is just explaining why you need to use a `@select` attribute on `<xsl:apply-templates>`.  

* You need to tell it to go and process the `<fs>` elements that have "Yes" answers in them.  So you need that `<xsl:apply-templates select=".....">`

* Then, after you CLOSE the template rule that processed the document root element, you need to write a NEW template rule to match on those `<fs>` elements, like this:

```
<xsl:template match="fs[the-ones-that-contain-Yes-answers]">
      <tr>
           <td><xsl:value-of select="count(...)"/><!--table row count here: count the preceding fs that has a Yes answer + 1 --></td>
           <td><!--question string here.  --> <xsl:apply-templates select="XPath-to-the-Question"/></td>
          <td> <!--Number of yes answers, pulled from an attribute value: you can use <xsl:apply-templates again or <xsl:value-of select="..."/> --></td>
       <!--  MORE <td> elements with <xsl:value-of select=".."> or <xsl:apply-templates select="..."/> to grab the data you want to go in the table cells.</td>-->
      </tr>
</xsl:template>

```

Does that help?

**Comment by Julegirl** [comment]
Kind of . . .

**Comment by Julegirl** [comment]
Looks like I'm running into a computer error. It says that I reached my maximum number of xsl:template instructions. I'm just going to turn in what I can and hopefully figure out where I went wrong tomorrow.

**Comment by ebeshero** [comment]
@Julegirl If you want to push your code here to the Troubleshooting folder, we can help debug it...Make sure you're starting your @select attribute on `<xsl:apply-templates>` using the descendant:: axis (no leading double-slash there!)

**Comment by Julegirl** [comment]
It's ok, I figured out why it was doing that. I'm good. And thanks for the 'no double-dash' tip. That was screwing me up.

**Comment by ebeshero** [comment]
Hooray for a successful debugging! You're learning some serious coding skills!!! :-D

**Comment by ebeshero** [comment]
@Julegirl That was one of my big mistakes in learning XSLT, too. I remember being really upset over the looping for quite a while, so I remember very vividly how to fix it as a result!!!

**Comment by Julegirl** [comment]
Thanks, but either way, I'll turn in what I can and figure it out tomorrow. I'm starting to lose brain power.
